nikola/nikola.py

```diff
@@ -1032,6 +1032,26 @@ def init_plugins(self, commands_only=False, load_all=False):
                 os.path.expanduser('~/.nikola/plugins'),
             ] + [utils.sys_encode(path) for path in extra_plugins_dirs if path]
 
+        compilers = defaultdict(set)
+        # Also add aliases for combinations with TRANSLATIONS_PATTERN
+        for compiler, exts in self.config['COMPILERS'].items():
+            for ext in exts:
+                compilers[compiler].add(ext)
+                for lang in self.config['TRANSLATIONS'].keys():
+                    candidate = utils.get_translation_candidate(self.config, "f" + ext, lang)
+                    compilers[compiler].add(candidate)
+
+        # Avoid redundant compilers
+        # Remove compilers that match nothing in POSTS/PAGES
+        # And put them in "bad compilers"
+        self.config['COMPILERS'] = {}
+        self.disabled_compilers = {}
+        self.disabled_compiler_extensions = defaultdict(list)
+        self.bad_compilers = set([])
+        for k, v in compilers.items():
+            # self.config['COMPILERS'][k] = sorted(list(v))
```

Comment:
Looks like a leftover from the original code. I've removed it and simplified the thing.